### DISCUSSION TITLE
Ecoworthy 5k inverter support

### DISCUSSION THREAD

**Discussion by ismaelnoble** [ai-republic/bms-to-inverter#62] [Q&A]
Good morning, 

I am stuck trying to get my ecoworthy 5k inverter connected. I have seen requests in the discussion section for srne inverter support, research into the unit shows it is a rebadged srne hyp4850s100-h. I am able to communicate with the unit using the srne solar station monitor 1.6  application and make changes. I have set the bms protocol to be pylon rs485 on both ends but the error message i get shows that the app is getting null response from the inverter.  I will post when i get home later but i want to think that the issue is that I am using the usb port and not the rs485 port on the inverter.  I havent been able to get the srne application to work with the rs485 port so there may be an issue there or the application does not work with the rs485 port.

**Comment by ismaelnoble** [comment]
good night, the error messages generated when attempting to connect to the inverter 
![image](https://github.com/user-attachments/assets/196e27be-093c-41d0-bb86-96ac217617e6)

**Comment by ismaelnoble** [comment]
Still no joy with getting communications up via serial port on the inverter, using the usb port on the inverter I am able to connect using the srne application with the settings below.
<img width="948" alt="Capture2" src="https://github.com/user-attachments/assets/e54d12aa-ef8f-4d14-8aa0-209c3d393562">

**Comment by ai-republic** [comment]
So it seems to get BMS data correctly. The sending fails, but I'm not quite sure yet why. Do you get data from `dev/ttyUSB1` using `cat /dev/ttyUSB0`?

**Comment by ismaelnoble** [comment]
/dev/ttyUSB0 is the Daly bms, I've mapped it to an alias on the system so the order doesn't change after reboot. I will try to connect to ttyusb1in a bit.

**Comment by ismaelnoble** [comment]
Ive connected to interface and got the below as a response;

![image](https://github.com/user-attachments/assets/d9171c01-4ece-42b7-964a-4193a6378271)

so we do have data just need the correct settings.

**Comment by ismaelnoble** [comment]
Still working on this, ive made some changes on the inverter and did a clean install earlier today and found a different set of errors this time. So great success or at least progress, screenshot below;

![image](https://github.com/user-attachments/assets/1c4bfd23-4395-4816-a4b1-cd9e45aa0942)

**Comment by ai-republic** [comment]
Hi @ismaelnoble,
hmm, the Daly BMS RS485 is reading the data correctly, but the issue with the inverter is that it can't find the device /dev/ttyUSB1:

![image](https://github.com/user-attachments/assets/7c2a1388-d813-4dc2-8ba7-cc0dd2c85bcb)

for your Pylon CAN inverter binding.
What type of USB CAN adapter are you using to connect to the inverter? Is it set up properly as CAN device?

**Reply by ismaelnoble** [reply]
the binding is setup as a pylon rs485. on the device it shows up as the same usb serial interface as the daly. screenshots below of the interface details from the linux machine.

inverter details:
![image](https://github.com/user-attachments/assets/78aef669-4985-47ce-8ea0-69826e599884)

Daly details:
![image](https://github.com/user-attachments/assets/441c036a-9af2-4631-84ba-989133f6c2b8)

fyi i do get coms via minicom and cat on the interface. It appears as what gibberish but is likely hex data:
![image](https://github.com/user-attachments/assets/a862db35-6b87-4053-8ef0-526e1eb56c2b)

**Reply by ismaelnoble** [reply]
on another note the inverter interface I am currently connected to is the built in usb interface. not the rs485 port, i still have not been able to correctly identify the correct pins to get coms up on the device. in my case i would be using a reg rj45 connector to a serial usb device. so identifying the correct pin on the rj45 to rj45 to db8 connector is hit or miss at the moment.

**Comment by ai-republic** [comment]
Hmm, in the console output I see this:
![image](https://github.com/user-attachments/assets/beadfd3b-5ce7-49cb-88c7-185d68d4b2f6)
which looks like you're using the Pylon CAN binding in the application and not the RS485. Please double check that 😉 
BMS data looks fine!

**Comment by ismaelnoble** [comment]
you were right i did set to pylon can at some point during messing with it. did a fresh install with pylon rs485 and got an out of bounds exception, screen shot below:
![image](https://github.com/user-attachments/assets/9245ee0f-c986-49bd-aed7-03ad86caf333)

**Comment by ai-republic** [comment]
Just pushed a fix for that. Please do a _Clean install_

**Comment by ismaelnoble** [comment]
Good Night;

 news and bad news, after the update the connection to the inverter is successfully made as shown below.
![image](https://github.com/user-attachments/assets/004c53f8-d276-4dac-ac34-839d3a0e9917)

however, onto the bad news. the data send fails.
![image](https://github.com/user-attachments/assets/3cb526ef-cbc8-4965-9edc-bfc1dc873d70)

![image](https://github.com/user-attachments/assets/26b55090-cbc5-4475-a9ba-3fef087a78c3)

i am leaning at it being an inverter issue but i could be wrong.

**Reply by ai-republic** [reply]
Could you attach the log file please (do not paste 😉 ) so that I can see the whole process. 
My guess is that your RS485 is not properly set up because the port could not be initialized and thus is _null_.
